src/tasks/WasmAppBuilder/WasmAppBuilder.cs

```diff
@@ -199,14 +218,29 @@ protected override bool ExecuteInternal()
         foreach (var assembly in _assemblies)
         {
             string assemblyPath = assembly;
-            if (UseWebcil)
-                assemblyPath = Path.ChangeExtension(assemblyPath, ".webcil");
-            config.Assets.Add(new AssemblyEntry(Path.GetFileName(assemblyPath)));
-            if (DebugLevel != 0) {
-                var pdb = assembly;
-                pdb = Path.ChangeExtension(pdb, ".pdb");
-                if (File.Exists(pdb))
-                    config.Assets.Add(new PdbEntry(Path.GetFileName(pdb)));
+            var bytes = File.ReadAllBytes(assemblyPath);
+            // for the is IL IsAssembly check we need to read the bytes from the original DLL
+            if (!IsAssembly(bytes))
+            {
+                Log.LogWarning("Skipping non-assembly file: " + assemblyPath);
```

Comment:
Log as a *message* right now, because it is not a test authoring error. It's an issue in our build where we don't correctly handle the publish outputs, and differentiate files. It will be fixed as part of fixing the build to work better for publish.